lib/posticle/src/lib.rs

```diff
@@ -88,10 +88,15 @@ pub fn extract_hashtags(text: &str, existing: &[Entity]) -> Vec<Entity> {
 /// which do not overlap with the `existing` ones.
 pub fn extract_mentions(text: &str, existing: &[Entity]) -> Vec<Entity> {
     regexes::RE_MENTION
-        .find_iter(text)
-        .map(|mat| Entity {
-            kind:  EntityKind::Mention,
-            range: (mat.start(), mat.end()),
+        .captures_iter(text)
+        .map(|capt| {
+            let whole = capt.get(0).unwrap();
+            let user = capt[1].to_string();
```

Comment:
Is there a big difference here between `.get.unwrap()` and `[]`?  It seems like they'll both panic if the capture group doesn't exist.

Reply:
`[]` returns an `&str`; `.get()` returns an `Option<Match<'t>>` which has the string indices of the start and end of that match group, which we use to construct the entity

Reply:
Interesting.  That seems like they both amount to slices (`&str` being defined as such and string indices being another way to return a slice of a string); I wonder what the rationale for that is.